opentelemetry-instrumentation/src/opentelemetry/instrumentation/_semconv.py

```diff
@@ -154,6 +171,23 @@ def _initialize(cls):
                 _OpenTelemetrySemanticConventionStability._OTEL_SEMCONV_STABILITY_SIGNAL_MAPPING[
                     _OpenTelemetryStabilitySignalType.HTTP
                 ] = http_opt_in
+
+                _OpenTelemetrySemanticConventionStability.server_duration_attrs_new_effective = (
+                    _server_duration_attrs_new_with_server_attributes
+                    if os.environ.get(
```

Comment:
We don't need to run this for all instrumentation migrated to new semconv, right? because not every instrumentation has metrics implemented. Wdyt we have a way to explicitly initialize this per instrumentation we want to support the opt-in attributes?

Reply:
+1 to supporting this individually per instrumentation. `_OpenTelemetrySemanticConventionStability` should only be used for semantic convention migration. We do not want to mix this functionality with semantic conventions.

Reply:
is this going in the right direction?

https://github.com/zeitlinger/opentelemetry-python-contrib/compare/http-metrics-server-arributes...zeitlinger:opentelemetry-python-contrib:http-metrics-server-arributes-asgi-trial?expand=1

Reply:
@zeitlinger 

Yeah you mostly like don't need to store the config as part of the global `_OpenTelemetrySemanticConventionStability `. I think you can store the config information in each instrumentation instance themselves when creating them. You can then have a static set of attributes with the one without the flag enabled and one with the flag enabled.

Reply:
I'm sympathetic to the @lzchen  approach. With that we can be more selective in which instrumentation we want to support to opt-in attributes

Reply:
ok - changed now :smile: